docs: native binaries installation docs (#442)
algorandfoundation/algokit-cli: README.md

## Prerequisites

AlgoKit has some runtime dependencies that also need to be available for particular commands.

> **Note**
> You can still install and use AlgoKit without these dependencies, and AlgoKit will tell you if you are missing one for a given command.

- Git - Git is used when creating and updating projects from templates
- Docker - Docker Compose (and by association, Docker) is used to run the AlgoKit LocalNet environment, we require Docker Compose 2.5.0+

> > **Note**
> > If you are using pipx to install AlgoKit, the key required dependency is Python 3.10+.
> 

Reviewer: ```suggestion
```

We already have this listed as a prerequisites in the pipx install instructions, so not need to have this note

> 

Reviewer: Maybe we should add a note, saying if you have previously installed AlgoKit using pipx, however would like to switch to using one of the other installations method, please uninstall using `pipx uninstall algokit`, before installing using the instructions for your desired platform.

## Cross-platform installation

> AlgoKit can be installed using OS specific package managers, or using the python tool [pipx](https://pypa.github.io/pipx/). See below for specific installation instructions.

Reviewer: Should the installation methods document the processor architectures we support?
One thing I did notice for windows was the binary built on the GitHub agent, seemed to work fine on my Windows ARM VM.

Author: the processor architectures we support are now documented. I stick to not supporting Windows ARM for now. If we think there is no issue supporting that then maybe we also need to change its release artifact name as well.

## Prerequisites

> AlgoKit has some runtime dependencies that also need to be available for particular commands.

Reviewer: ```suggestion
Your specific installation method documents the prerequisites for running AlgoKit. Additionally AlgoKit has some runtime dependencies that also need to be available for particular commands.
```

> **Note**
> You can still install and use AlgoKit without these dependencies, and AlgoKit will tell you if you are missing one for a given command.

- Git - Git is used when creating and updating projects from templates
- Docker - Docker Compose (and by association, Docker) is used to run the AlgoKit LocalNet environment, we require Docker Compose 2.5.0+

> **Note**
> If you have previously installed AlgoKit using `pipx` and would like to switch to a different installation method, please ensure that 
> you first uninstall the existing version by running `pipx uninstall algokit`. Once uninstalled, you can follow the installation instructions for your preferred platform.

## Cross-platform installation

AlgoKit can be installed using OS specific package managers, or using the python tool [pipx](https://pypa.github.io/pipx/). 
See below for specific installation instructions and supported architectures.

### Supported Operating Systems and Architectures

- **macOS**: x64 and ARM (Apple Silicon)

> - **Windows**: Windows 10 and later (x64 only; ARM not supported)

Reviewer: WinGet is only support on `Windows 10 1709 (build 16299)` and later.

## Install AlgoKit on Windows

1. Ensure prerequisites are installed

>    - [winget](https://learn.microsoft.com/en-us/windows/package-manager/winget/)

Reviewer: ```suggestion
   - [WinGet](https://learn.microsoft.com/en-us/windows/package-manager/winget/) (should be installed by default on recent Windows 10 or later)
```

   - [Git](https://github.com/git-guides/install-git#install-git-on-windows) (or `winget install git.git`)
   - [Docker](https://docs.docker.com/desktop/install/windows-install/) (or `winget install docker.dockerdesktop`)
     > **Note**
     > See [our LocalNet documentation](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/features/localnet.md#prerequisites) for more tips on installing Docker on Windows
     

> 2. Install using winget

Reviewer: ```suggestion
2. Install using WinGet
```

    ```shell
    winget install algokit
    ```

3. [Verify installation](#verify-installation)

### Maintenance

Some useful commands for updating or removing AlgoKit in the future.

> - To update AlgoKit: `winget upgrade AlgoKit`

Reviewer: ```suggestion
- To update AlgoKit: `winget upgrade algokit`
```

> - To remove AlgoKit: `winget uninstall AlgoKit`

Reviewer: ```suggestion
- To remove AlgoKit: `winget uninstall algokit`
```

1. Ensure prerequisites are installed

>    - [Snap](https://snapcraft.io/docs/installing-snapd) (should already be available on Ubuntu 16.04.4 LTS (Xenial Xerus) or later)

Reviewer: ```suggestion
   - [Snap](https://snapcraft.io/docs/installing-snapd) (should be installed by default on Ubuntu 16.04.4 LTS (Xenial Xerus) or later)
```
